sky/packages/sky/lib/widgets/date_picker.dart

```diff
@@ -0,0 +1,392 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:sky/theme/colors.dart' as colors;
+import 'package:sky/theme/typography.dart' as typography;
+import 'package:sky/widgets.dart';
+import 'package:intl/intl.dart';
+import 'package:intl/date_symbols.dart';
+
+typedef void DatePickerValueChanged(DateTime dateTime);
+
+enum DatePickerMode { day, year }
+
+typedef void DatePickerModeChanged(DatePickerMode value);
+
+class DatePicker extends StatefulComponent {
+  DatePicker({
+    this.selectedDate,
+    this.onChanged,
+    this.firstDate,
+    this.lastDate
+  }) {
+    assert(selectedDate != null);
+    assert(firstDate != null);
+    assert(lastDate != null);
+  }
+
+  DateTime selectedDate;
+  DatePickerValueChanged onChanged;
+  DateTime firstDate;
+  DateTime lastDate;
+
+  void syncConstructorArguments(DatePicker source) {
+    selectedDate = source.selectedDate;
+    onChanged = source.onChanged;
+    firstDate = source.firstDate;
+    lastDate = source.lastDate;
+  }
+
+  DatePickerMode _mode = DatePickerMode.day;
+
+  void _handleModeChanged(DatePickerMode mode) {
+    setState(() {
+      _mode = mode;
+    });
+  }
+
+  void _handleYearChanged(DateTime dateTime) {
+    setState(() {
+      _mode = DatePickerMode.day;
+    });
+    if (onChanged != null)
+      onChanged(dateTime);
+  }
+
+  static const double calendarHeight = 210.0;
+
+  Widget build() {
+    Widget header = new DatePickerHeader(
+      selectedDate: selectedDate,
+      mode: _mode,
+      onModeChanged: _handleModeChanged
+    );
+    Widget picker;
+    switch (_mode) {
+      case DatePickerMode.day:
+        picker = new MonthPicker(
+          selectedDate: selectedDate,
+          onChanged: onChanged,
+          firstDate: firstDate,
+          lastDate: lastDate,
+          itemExtent: calendarHeight
+        );
+        break;
+      case DatePickerMode.year:
+        picker = new YearPicker(
+          selectedDate: selectedDate,
+          onChanged: _handleYearChanged,
+          firstDate: firstDate,
+          lastDate: lastDate
+        );
+        break;
+    }
+    return new BlockBody([header, new Container(height: calendarHeight, child: picker)]);
```

Comment:
Column wasn't working for me, I'll try to make a reduced test case for @Hixie to look at